controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/routing/RoutingApiHandler.java

```diff
@@ -85,11 +86,38 @@ private HttpResponse get(Path path, HttpRequest request) {
         if (path.matches("/routing/v1/status/tenant/{tenant}/application/{application}")) return application(path, request);
         if (path.matches("/routing/v1/status/tenant/{tenant}/application/{application}/instance/{instance}")) return instance(path, request);
         if (path.matches("/routing/v1/status/tenant/{tenant}/application/{application}/instance/{instance}/environment/{environment}/region/{region}")) return deployment(path);
+        if (path.matches("/routing/v1/status/tenant/{tenant}/application/{application}/instance/{instance}/endpoint")) return endpoints(path);
         if (path.matches("/routing/v1/status/environment")) return environment(request);
         if (path.matches("/routing/v1/status/environment/{environment}/region/{region}")) return zone(path);
         return ErrorResponse.notFoundError("Nothing at " + path);
     }
 
+    private HttpResponse endpoints(Path path) {
+        var instanceId = instanceFrom(path);
+        var endpoints = controller.routing().endpointsOf(instanceId);
+        var zoneStatus = endpoints.asList().stream()
+                .flatMap(e -> e.zones().stream())
+                .distinct()
+                .collect(Collectors.toMap(
+                        zone -> zone,
+                        zone -> controller.routing().globalRotationStatus(new DeploymentId(instanceId, zone))
```

Comment:
This will only get status for endpoints having rotations. Have a look at this method: https://github.com/vespa-engine/vespa/blob/a8c616b18803e2e3a959b9ba987deb64fabf252a/controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/routing/RoutingApiHandler.java#L258-L306

Try extracting a method from this and reuse it here.